src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -316,11 +337,45 @@ public void validateTimestamp(QueryState queryState, QueryOptions options)
         Guardrails.minimumAllowableTimestamp.guard(ts, table(), false, queryState.getClientState());
     }
 
+    public void validateConstraints()
```

Comment:
Constraints are only applicable for inserts/updates. `ModificationStatement` applies to deletion too.

Asking 1. Can you relocate the validation to `UpdateStatement.java` file?

Asking 2. Constraints validation can be performed easier before `execute`, during `prepare`, where it has all the bound values and corresponding column metadata. The constraints should be validated before an `UpdateStatement` can be created.